src/plugins/browse/mod.rs

```diff
@@ -59,66 +72,98 @@ impl Error for BrowseNoArgError {}
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct NoContentError {
     error: String,
-    help: String
+    help: String,
 }
 
+// a function that chunks text into fixed sizes, measured by the count of tokens (bpe)
 fn chunk_text(text: &str, chunk_size: usize) -> Vec<String> {
+    let bpe = cl100k_base().unwrap();
+    let tokens = bpe.encode_ordinary(text);
     let mut chunks = vec![];
-    let mut current_chunk = String::new();
+    let mut token_vec: Vec<_> = tokens.to_vec();
 
-    for word in wrap(text, chunk_size) {
-        if current_chunk.len() + word.len() > chunk_size {
-            chunks.push(current_chunk.trim().to_owned());
-            current_chunk = String::new();
-        }
-        current_chunk.push_str(&word);
-    }
-    if !current_chunk.is_empty() {
-        chunks.push(current_chunk.trim().to_owned());
+    while !token_vec.is_empty() {
+        let drain_to = std::cmp::min(token_vec.len(), chunk_size);
+        let token_chunk = token_vec.drain(0..drain_to).collect::<Vec<_>>();
+
+        let chunk = bpe.decode(token_chunk).unwrap();
+        chunks.push(chunk);
```

Comment:
this chunk_text method is based on counting tokens with tiktoken library, which the project already uses

Reply:
This isn't ideal because different models tokenize in different ways. Ideally, we should implement per-model tokenization and use that.